prober/grpc.go

```diff
@@ -0,0 +1,55 @@
+package prober
+
+import (
+	"context"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/prometheus/blackbox_exporter/config"
+
+	"google.golang.org/grpc"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+)
+
+func ProbeGRPC(ctx context.Context, target string, module config.Module, registry *prometheus.Registry, logger log.Logger) bool {
+	var (
+		conn    *grpc.ClientConn
+		parts   []string
+		service string
+		host    string
+	)
+
+	parts          = strings.Split(target, "/")
+	host, service  = parts[0], parts[1]
+	conn, err     := grpc.Dial(host, grpc.WithInsecure())
+
+	if err != nil {
+		level.Error(logger).Log("msg", "failed to connect")
+		return false
+	}
+
+	defer conn.Close()
+
+	level.Error(logger).Log("msg", "invoking", "host", host, "server", service, "method", module.GRPC.Method, "request", healthpb.HealthCheckRequest{Service: service})
+
+	req := &healthpb.HealthCheckRequest{Service: service}
```

Comment:
This prober type is an odd mixture of a generic grpc prober and a grpc HealthCheck service prober. I think it should be one or the other entirely. If this is going to send `HealthCheckRequest` payloads and expect a `HealthCheckResponse` response, then there is no need for the grpc method. In that case, I'd suggest the probe should be called grpc_health or something.

Or you go completely generic and somehow allow the probe config to specify the payload (and what to look for in the response). That's going to be tricky.